Avoid overwriting default locale

rkworks opened the pull request:
This PR is related to #210.
 
That change causes weird behavior in my environment.
If "ja_JP" (my default locale)  is passed to DocEnv/setLocale, the default locale will be overwritten to "ja_JP_JP_#u-ca-japanese".
When new locale is passed to DocEnv/setLocale again, it becomes the error. 

I've read source code of com.sun.tools.javadoc.DocLocale and found that `getLocale` method returns the default locale itself when passed locale name is empty string.

```clj
user=> (require 'cider.nrepl.middleware.util.java.parser)
nil
user=> (in-ns 'cider.nrepl.middleware.util.java.parser)
#<Namespace cider.nrepl.middleware.util.java.parser>
cider.nrepl.middleware.util.java.parser=> (def docenv (let [access   (ModifierFilter. ModifierFilter/ALL_ACCESS)
                                     #_=>                   context  (doto (Context.) (Messager/preRegister "cider-nrepl-javadoc"))
                                     #_=>                   options  (doto (Options/instance context) (.put "ignore.symbol.file" "y"))
                                     #_=>                   compiler (JavadocTool/make0 context)
                                     #_=>                   enter    (JavadocEnter/instance0 context)]
                                     #_=>               (DocEnv/instance context)))
#'cider.nrepl.middleware.util.java.parser/docenv
cider.nrepl.middleware.util.java.parser=> (Locale/setDefault (Locale. "ja" "JP"))
nil
cider.nrepl.middleware.util.java.parser=> (Locale/getDefault)
#<Locale ja_JP>
cider.nrepl.middleware.util.java.parser=> (.setLocale docenv (str (Locale/getDefault)))
nil
cider.nrepl.middleware.util.java.parser=> (Locale/getDefault)
#<Locale ja_JP_JP_#u-ca-japanese>
cider.nrepl.middleware.util.java.parser=> (.setLocale docenv (str (Locale/getDefault)))
cider-nrepl-javadoc: エラー - ロケールja_JP_JP_#u-ca-japaneseが無効です

ExitJavadoc   com.sun.tools.javadoc.Messager.exit (Messager.java:300)
cider.nrepl.middleware.util.java.parser=> (Locale/setDefault (Locale. "ja" "JP"))
nil
cider.nrepl.middleware.util.java.parser=> (.setLocale docenv "")
nil
cider.nrepl.middleware.util.java.parser=> (Locale/getDefault)
#<Locale ja_JP>
```

expez commented:
Would you mind adding some of this explanation to the commit body--or better yet as a code comment--so it's available when another developer think's the `""` is a code smell and either deletes it as a no-op or changes it do be the seemingly harmless, but arguably more readable,  `(Locale/getDefault)`?

rkworks commented:
Ah, sorry, I hadn't read all discussions on #210. 
As @cursork said, default locale name is empty string , so all I need is just remove the setLocale call. I'm not sure whether this can cause unintended behavior, but the current version is broken at least in my environment...

Anyway, I've changed the code and added some explanation to the commit body.

expez commented:
As I said in the other issue I think this is a case where we just have to try it out and see what happens.  Writing tests for stuff like this is super hard.

Weird that the coveralls check is failing, and that it indicates decreased coverage.  The amount of tests are the same but we have one line less code.  The coverage should be slightly up :)

rkworks commented:
Thanks! I hope nothing will happen... :p

haha, I've just wondered too

expez commented:
New snapshot is out, if you want to enjoy the fruits of your labor immediately :) 

Thanks! :+1:
